plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotTreeUtilities.java

```diff
@@ -227,4 +234,39 @@ public static SWTBotTreeItem select(SWTWorkbenchBot bot, SWTBotTree tree, String
     }
     return item.getNode(nodeNames[leafIndex]).select(); // throws WNFE
   }
+
+  /** Expand the tree as necessary to find a child matching the given condition. */
+  public static boolean hasChild(SWTWorkbenchBot bot, SWTBotTree tree, Matcher<String> textMatcher) {
+    waitUntilTreeHasItems(bot, tree);
+    // perform breadth-first search; execute directly in SWT thread as we can
+    // be affected by tree changes due to thread changes
+    Result<Boolean> query =
+        () -> {
+          TreeItem[] items = tree.widget.getItems();
+          for (TreeItem item : items) {
+            System.out.println("Debug Item: " + item.getText());
+            if (textMatcher.matches(item.getText())) {
+              return true;
+            }
+          }
+          LinkedList<TreeItem> stack = new LinkedList<>();
+          Collections.addAll(stack, items);
+          while (!stack.isEmpty()) {
+            TreeItem parent = stack.removeFirst();
+            items = parent.getItems();
+            assertFalse(
+                "need workaround for https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2569",
```

Comment:
It took me a while to follow this. I think it's better to have a code comment, because when I see a message in `assert`, I usually assume it explains the reason for a legitimate bug or what went wrong or what should be the state to be correct. But here, it is not really a bug to fix, but some workaround to make the test work. So, something like

```
// If this assertion fails,  it may be due to https://... Apply the workaround to collapse and re-expand the item (and adjust this comment accordingly.)
assertFalse(items.length == 1 && ...)
```

Reply:
It's not a workaround: it's there to detect whether the situation described in that issue occurs. I'll revise the code though to make it clear.